device.mk

```diff
@@ -205,4 +205,26 @@ PRODUCT_PROPERTY_OVERRIDES += \
 #
 PRODUCT_DEFAULT_PROPERTY_OVERRIDES += \
     persist.sys.usb.config=mtp \
```

Comment:
I know that the PID changes (each USB devices has a VID:PID) when you change modes. So MTP will have a slightly different PID from mass_storage, which will be different again when adb is enabled or not).

If you're running in a virtual machine, this can impact the filtering.

You can see this in the /init.qcom.usb.rc file.

Make sure that there is an entry in that file for the mass_storage and mass_storage with adb settings. Its quite possible it doesn't exist.